pkg/pillar/hypervisor/xen.go

```diff
@@ -586,6 +586,11 @@ func (ctx xenContext) PCIRelease(long string) error {
 	return nil
 }
 
+func (ctx xenContext) PCISameController(id1 string, id2 string) bool {
+	// we maybe better off returning types.PCISameController(id1, id2) here
```

Comment:
Yes I don't think we need to change the behavior for Xen.

Reply:
I disagree @eriknordmark  -- if you disable the check, you won't be able to have a parity between Xen and KVM on the boxes with non-trivial iommu groups and until we deprecate Xen completely that's a much bigger issue for our support. So unless you have some other way of solving this -- we currently have a regression in functionality.

Xen actually does the right thing when you try to assign incorrect iommu groups to DomU -- so that's not an issue. The only issue, thus, is assigning something to pciback when it is supposed to be available to Dom0. We don't (and we absolutely won't ever) have an ironclad way of avoiding that. Even the KVM check will actually fail from time to time (since we're running KVM in permissive mode) -- hence this is belt-n-suspenders and we still should really avoid putting incorrect models into the field.

Reply:
Based on our conversation, I think we can remove the suggestion from the above comment and instead say something like:
 We can not currently do enforcement based on iommu groups for Xen, since the hypervisor hides that from dom0.Thus we assume that initial bringup and model creation is done using KVM. That model can then be used with KVM and Xen.

Reply:
Done